nemo/collections/vlm/neva/data/lazy.py

```diff
@@ -0,0 +1,622 @@
+        tokens = tokenize_special_token(conversations, self.tokenizer)
+        labels = torch.ones_like(tokens) * IGNORE_INDEX
+        search_start_index = 0
+        for i in range(1, len(self.conv.messages), 2):
+            stop_str = getattr(self.conv, "stop_str", None)
+            assert (
+                stop_str is not None
+            ), "If `stop_str` is not provided, issues might occur in labeling the answer tokens."
+            answer_tokens = self.tokenizer.encode(
+                self.conv.messages[i][1] + ("" if stop_str is None else stop_str),
+                add_special_tokens=False,
+                return_tensors="pt",
+            )[0]
+            answer_start, answer_end = find_pattern_indices(tokens, answer_tokens, search_start_index)
+            labels[answer_start:answer_end] = tokens[answer_start:answer_end]
+            search_start_index = answer_end
+        tokens = tokens[:-1]
+        labels = labels[1:]
+        return tokens, labels
+
+    def _get_crop_size(self):
+        if isinstance(self.image_processor, CLIPImageProcessor):
+            return [self.image_processor.crop_size['height'], self.image_processor.crop_size['width']]
+        else:
+            raise NotImplementedError
+
+
+class NevaDataset(LazySupervisedDataset):
+    """Dataset for supervised fine-tuning."""
+
+    def __init__(
+        self,
+        data_path,
+        data_config,
+        tokenizer,
+        image_processor,
+    ):
+
+        if data_path.endswith(".json"):
+            super().__init__(data_path, data_config, tokenizer, image_processor)
+
+        elif data_path.endswith(".jsonl"):
+            super().__init__(None, data_config, tokenizer, image_processor)
+            logging.warning("Loading image inputs from SteerLM Dataset...")
+            if data_config.media_type == 'image':
+                image_folder = data_config.image_folder
+                for line in open(data_path, "r"):
+                    record = json.loads(line)
+
+                    # This currently supports only a single image
+                    # search for <img src="/absolute/path/to/image" in the conversation
+                    #   add it as record['image'], remove src tag from the <img> tag
+
+                    record['image'] = []
+                    for turn in record['conversations']:
+                        matches = re.finditer('<img src="([^"]+)"', turn['value'])
+                        for match in matches:
+                            image_name = match.group(1).split("/")[-1]
+                            image_path = os.path.join(image_folder, image_name)
+                            if not os.path.isfile(image_path):
+                                logging.warning(f"Image not found: {image_path}")
+                                continue
+                            record['image'].append(image_name)  # url
+                        turn['value'] = re.sub('<img src="([^"]+)">', IMAGE_TOKEN_INDEX, turn['value'])
```

Comment:
IMAGE_TOKEN_INDEX is integer type, cannot be directly subbed into string

Reply:
change to "<image>"